pytorch_lightning/trainer/connectors/logger_connector/logger_connector.py

```diff
@@ -158,6 +159,9 @@ def configure_logger(self, logger):
             else:
                 self.trainer.logger = logger
 
+        # connect trainer to logger
+        logger.connect(self.trainer)
```

Comment:
can we maybe already make it a weakproxy here? We don't know if all the loggers will correctly have only a weakref on this otherwise.
```suggestion
        logger.connect(weakref.proxy(self.trainer))
```